gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/dichotomy/DichotomyParallelization.java

```diff
@@ -6,75 +6,70 @@
-                // .and(() -> runDichotomyForOneDirection(sweData, Direction.PT_ES))
-                .close();
+        ExecutionResult<SweDichotomyResult> executionResult = runAndGetSweDichotomyResults(sweData);
         dichotomyLogging.logEndAllDichotomies();
         String ttcDocUrl = outputService.buildAndExportTtcDocument(sweData, executionResult);
-        String voltageEsFrZipUrl = outputService.buildAndExportVoltageDoc(DichotomyDirection.ES_FR, sweData, executionResult);
         SweDichotomyResult esFrResult = getDichotomyResultByDirection(executionResult, DichotomyDirection.ES_FR);
-        return new SweResponse(sweData.getId(), ttcDocUrl, voltageEsFrZipUrl, esFrResult.getHighestValidStepUrl(), esFrResult.getLowestInvalidStepUrl(), esFrResult.getExportedCgmesUrl());
+        return new SweResponse(sweData.getId(), ttcDocUrl, esFrResult.getHighestValidStepUrl(), esFrResult.getLowestInvalidStepUrl(), esFrResult.getExportedCgmesUrl());
     }
 
-    SweDichotomyResult runDichotomyForOneDirection(SweData sweData, DichotomyDirection direction) {
-        // propagate in logs MDC the task requestId as an extra field to be able to send logs with calculation tasks.
-        MDC.put("gridcapa-task-id", sweData.getId());
-        DichotomyResult<RaoResponse> dichotomyResult = dichotomyRunner.run(sweData, direction);
-        dichotomyLogging.logEndOneDichotomy(direction);
-        String zippedCgmesUrl = cgmesExportService.buildAndExportCgmesFiles(direction, sweData, dichotomyResult);
-        String highestValidStepUrl = cneFileExportService.exportCneUrl(sweData, dichotomyResult, true, direction);
-        String lowestInvalidStepUrl = cneFileExportService.exportCneUrl(sweData, dichotomyResult, false, direction);
-        Optional<VoltageMonitoringResult> voltageMonitoringResult = voltageCheckService.runVoltageCheck(sweData, dichotomyResult, direction);
-        dichotomyLogging.generateSummaryEvents(direction, dichotomyResult, sweData, voltageMonitoringResult);
-        return new SweDichotomyResult(direction, dichotomyResult, voltageMonitoringResult, zippedCgmesUrl, highestValidStepUrl, lowestInvalidStepUrl);
+    private ExecutionResult<SweDichotomyResult> runAndGetSweDichotomyResults(SweData sweData) {
+        ExecutionResult<SweDichotomyResult> executionResult = null;
+        CompletableFuture<SweDichotomyResult> dichoEsFr = worker.runDichotomyForOneDirection(sweData, DichotomyDirection.ES_FR);
+        CompletableFuture<SweDichotomyResult> dichoFrEs = worker.runDichotomyForOneDirection(sweData, DichotomyDirection.FR_ES);
+        CompletableFuture<SweDichotomyResult> dichoEsPt = worker.runDichotomyForOneDirection(sweData, DichotomyDirection.ES_PT);
+        CompletableFuture<SweDichotomyResult> dichoPtEs = worker.runDichotomyForOneDirection(sweData, DichotomyDirection.PT_ES);
+        List<SweDichotomyResult> results = new ArrayList<>();
+        waitAndAddToReults(dichoEsFr, results, DichotomyDirection.ES_FR);
```

Comment:
I dont really like using pointers to add elements to a list. Maybe the method waitAndAddToReults could return an element that you set in a results.add(). TBD

Reply:
ok